### PR TITLE
fix: wrong cast of !Ref when using CommaDelimitedList

### PR DESCRIPTION
**Pull request by EloB**
#### Which issue(s) does this change fix?

#4026

#### Why is this change necessary?

When you pass a parameter with the type `CommaDelimitedList` and use it with `!Ref` it should be transformed to an array of strings. Right now it's impossible to set `AllowOrigins` in local development with `!Ref`. You always get "CORS Property AllowOrigins was not fully resolved. Will proceed as if the Property was not defined." when using `sam local start-api`.

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/parameters-section-structure.html#parameters-section-structure-properties

> CommaDelimitedList
> An array of literal strings that are separated by commas. The total number of strings should be one more than the total number of commas. Also, each member string is space trimmed.
> 
> For example, users could specify "test,dev,prod", and a Ref would result in ["test","dev","prod"].

#### How does it address the issue?

Resolves the correct type.

#### What side effects does this change have?

Don't know. First time I touched python. I'm happy that I got this far and fixed the issue. Review with care. Gladly take any feedback on how todo this the proper "python way". Best would probably if you guys take over this. 😅 

I've tried this using `samdev local start-api` and was able to enable CORS using this fix.

#### Mandatory Checklist
**PRs will only be reviewed after checklist is complete**

- [ ] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [ ] Write design document if needed ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [ ] Write/update unit tests
- [ ] Write/update integration tests
- [ ] Write/update functional tests if needed
- [x] `make pr` passes
- [ ] `make update-reproducible-reqs` if dependencies were changed
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

**Comment by EloB** [comment]
I've tried this using `samdev local start-api` and was able to enable CORS using this fix.

**Comment by hnnasit** [comment]
Hi @EloB, thanks for your contributions and the detailed information :+1:. We will try to validate the changes to make sure everything is working as expected.

**Comment by sidhujus** [comment]
Hi @EloB thanks for your contribution! It looks great. Could you add some unit tests to cover the new logic? Thanks

File: samcli/lib/intrinsic_resolver/intrinsics_symbol_table.py [hunk 1]
```diff
@@ -324,6 +324,9 @@ def get_translation(self, logical_id, resource_attributes=IntrinsicResolver.REF)
         if any(isinstance(logical_id_item, object_type) for object_type in [str, list, bool, int]):
             if resource_attributes not in (IntrinsicResolver.REF, ""):
                 return None
+            parameter_info = self._parameters.get(logical_id)
+            if parameter_info and parameter_info.get("Type") == "CommaDelimitedList":
```
Comment (sidhujus) [inline samcli/lib/intrinsic_resolver/intrinsics_symbol_table.py:328]: Can we move this to a constant or an enum

**Comment by EloB** [comment]
@sidhujus @hnnasit I think it's better if you guys can help finalizing this issue. I don't have the experience to be able to write unit test and best practice syntax. The only knowledge I have in Python right now was to enable the debugger and step through the code than I let ChatGPT write the fix.

Best regards :)

**Comment by EloB** [comment]
Would be good if this becomes merged because it fixes a bug/broken functionality. Right now it blocks local development if you have CORS functionality.

**Review by sidhujus — APPROVED** [review]

**Review by hnnasit — APPROVED** [review]

**Comment by mildaniel** [comment]
Thank you for the contribution @EloB!

**Pull request merged**